File: backend/src/main/java/org/bootstrapbugz/api/shared/error/exception/ConflictException.java
package org.bootstrapbugz.api.shared.error.exception;

import java.io.Serial;
import lombok.Getter;
import org.springframework.http.HttpStatus;

@Getter
public class ConflictException extends RuntimeException {
  @Serial private static final long serialVersionUID = 8841655774286844538L;
  private final HttpStatus status;
  private final String field;

  public ConflictException(String field, String message) {
    super(message);
    this.status = HttpStatus.CONFLICT;
    this.field = field;
  }
}

File: backend/spring-boot-kotlin/src/main/kotlin/org/bootstrapbugz/api/shared/error/exception/ConflictException.kt
package org.bootstrapbugz.api.shared.error.exception

import java.io.Serial
import org.springframework.http.HttpStatus

class ConflictException : RuntimeException {
  val status: HttpStatus
  val field: String?

  constructor(field: String?, message: String?) : super(message) {
    status = HttpStatus.CONFLICT
    this.field = field
  }

  constructor(message: String?) : super(message) {
    status = HttpStatus.CONFLICT
    field = null
  }

  companion object {
    @Serial private val serialVersionUID = 8841655774286844538L
  }
}
